src/main/java/code401challenges/linkedlist/LinkedList.java

```diff
@@ -56,4 +51,67 @@ public String toString(){
         }
         return allValues.toString();
     }
+
+
+
+    public void append(int value){
+
+    }
+
+
+
+    public void insertBefore(int value,int newVal){
+//        check to make sure the list has stuff in it
+        if(head == null || head.next == null){
+            System.out.println("Sad eagle screams in sorrow :( the action is not possible.");
+        }else if(head.value == value){
+            prepend(newVal);
+        }
+        else{
+//       Track node current and next looking for the right place to insert
+//            instantiate here as prepend will make a new node and place at beginning if it is called
+            Node newNode = new Node(newVal);
+
+            Node prevNode = head;
+            Node currNode = prevNode.next;
+
+            while (currNode != null && currNode.value != value){
+                prevNode = prevNode.next;
+                currNode = currNode.next;
+            }
+            if(currNode == null){
+                System.out.println("Sad eagle screams in sorrow :( the action is not possible.");
+            }else{
+                newNode.next = currNode;
+                prevNode.next = newNode;
+            }
+        }
+    }
+
+
+
+    public void insertAfter(int value,int newVal) {
+//       create a new node to insert
+        Node newNode = new Node(newVal);
+//        check to make sure the list has stuff in it
+        if(head == null || head.next == null){
+            System.out.println("Sad eagle screams in sorrow :( the action is not possible.");
+        }
+        else{
+//            Track
+            Node currNode = head;
+            Node nextNode =currNode.next;
+            while (nextNode != null && currNode.value != value){
+                currNode = currNode.next;
+                nextNode = nextNode.next;
+            }
+            if(nextNode == null){
+                System.out.println("Sad eagle screams in sorrow :( the action is not possible.");
+            }else{
+
+                newNode.next = nextNode;
+                currNode.next = newNode;

```

Comment:
The nextNode might be set to null if you are inserting after the last node in the list, but that should be ok because if you insert a node at the end, the set its next with nextNode (null), then it has been set up correctly.